### ISSUE TITLE
Bug: Gradient Accumulation

### ISSUE THREAD

**Issue by geezah** [Lightricks/LTX-Video-Trainer#46]
First of all: Thank you for your efforts in democratizing the training of a state-of-the-art diffusion pipeline! As many official codebases only provide code and scripts for inference, this is really helpful.

In the training loop located at `trainer.py:196-289`, the gradient accumulation is not correctly implemented. Lines 213-227 read:

```python3
is_optimization_step = (step + 1) % cfg.optimization.gradient_accumulation_steps == 0
if is_optimization_step:
    self._global_step += 1

loss = self._training_step(batch)
self._accelerator.backward(loss)

if self._accelerator.sync_gradients and cfg.optimization.max_grad_norm > 0:
    self._accelerator.clip_grad_norm_(
        self._trainable_params,
        cfg.optimization.max_grad_norm,
    )

self._optimizer.step()
self._optimizer.zero_grad()
```
The step function of the optimizer is called at each step. Hence, accumulation does not take place here. Let me know if I am missing something here or have identified this correctly.

**Comment by geezah** [comment]
Ah, got it - thanks for the quick response!
